main/src/main/java/org/mobilitydata/gtfsvalidator/validator/ShapeIncreasingDistanceValidator.java

```diff
@@ -47,33 +57,121 @@ public void validate(NoticeContainer noticeContainer) {
       for (int i = 1; i < shapeList.size(); ++i) {
         GtfsShape prev = shapeList.get(i - 1);
         GtfsShape curr = shapeList.get(i);
-        if (prev.hasShapeDistTraveled()
-            && curr.hasShapeDistTraveled()
-            && prev.shapeDistTraveled() >= curr.shapeDistTraveled()) {
+        if (!(prev.hasShapeDistTraveled() || curr.hasShapeDistTraveled())) {
+          continue;
+        }
+        if (prev.shapeDistTraveled() > curr.shapeDistTraveled()) {
+          noticeContainer.addValidationNotice(new DecreasingShapeDistanceNotice(prev, curr));
+          continue;
+        }
+        if (prev.shapeDistTraveled() != curr.shapeDistTraveled()) {
+          continue;
+        }
+        // equal shape_dist_traveled and different coordinates
+        if (curr.shapePtLat() != prev.shapePtLat() || curr.shapePtLon() != prev.shapePtLon()) {
+          // check if shape points are more than 1.11 meters away
+          if (MAX_DISTANCE_SHAPEPOINTS_METERS
+              < getDistanceMeters(curr.shapePtLatLon(), prev.shapePtLatLon())) {
+            noticeContainer.addValidationNotice(
+                new EqualShapeDistanceDiffCoordinatesNotice(prev, curr));
+          }
+        } else if (curr.shapePtLat() == prev.shapePtLat()
+            && curr.shapePtLon() == prev.shapePtLon()) {
+          // equal shape_dist_traveled and same coordinates
           noticeContainer.addValidationNotice(
-              new DecreasingOrEqualShapeDistanceNotice(
-                  curr.shapeId(),
-                  curr.csvRowNumber(),
-                  curr.shapeDistTraveled(),
-                  curr.shapePtSequence(),
-                  prev.csvRowNumber(),
-                  prev.shapeDistTraveled(),
-                  prev.shapePtSequence()));
+              new EqualShapeDistanceSameCoordinatesNotice(prev, curr));
```

Comment:
What if coordinates are different and `getDistanceMeters(curr.shapePtLatLon(), prev.shapePtLatLon()) <= MAX_DISTANCE_SHAPEPOINTS_METERS `?

Reply:
In this case, no notice should be triggered, see: https://github.com/MobilityData/gtfs-validator/blob/737adabd6a2358cdd5f470e741d67149db7e3184/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/ShapeIncreasingDistanceValidatorTest.java#L103